app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt

```diff
@@ -139,20 +124,25 @@ class AdminPinActivityPresenter @Inject constructor(
         }
       )
     }
-  }
 
-  private fun addTextChangedListener(
-    profileInputView: ProfileInputView,
-    onTextChanged: (CharSequence?) -> Unit
-  ) {
-    profileInputView.addTextChangedListener(object : TextWatcher {
-      override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
-      override fun afterTextChanged(p0: Editable?) {
-        onTextChanged(p0)
+    binding.adminPinInputConfirmPinEditText.setOnEditorActionListener { _, actionId, event ->
+      if (actionId == EditorInfo.IME_ACTION_DONE ||
+        (event != null && (event.keyCode == KeyEvent.KEYCODE_ENTER))
+      ) {
+        binding.submitButton.callOnClick()
       }
-
-      override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
-    })
+      false
+    }
+    binding.adminPinInputPinEditText.setOnEditorActionListener { _, actionId, event ->
+      if (actionId == EditorInfo.IME_ACTION_DONE ||
+        (event != null && (event.keyCode == KeyEvent.KEYCODE_ENTER))
+      ) {
+        binding.submitButton.callOnClick()
+      }
+      false
+    }
+    binding.adminPinInputPinEditText.setText(adminViewModel.savedPin.get().toString())
+    binding.adminPinInputConfirmPinEditText.setText(adminViewModel.savedConfirmPin.get().toString())
```

Comment:
earlier we use `setInput` function from `ProfileInputView` but after the migration can we perform text setting thing via data binding!!
@prayutsu @FareesHussain WDYT?

Reply:
@anandwana001 we have to do two-way data binding here??

Reply:
@anandwana001 Using data binding is a better practice, so we should definitely use it.